src/systems.jl

```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
+- `U` -- input constraints
+"""
+SecondOrderConstrainedAffineControlContinuousSystem
+
+@doc """
+    SecondOrderConstrainedAffineControlDiscreteSystem
+
+Discrete-time second order constrained affine control system of the form:
+
+```math
+    Mx_{k+2} + Cx_{k+1} + Kx_k = Bu_k + d, \\; x_k ∈ \\mathcal{X}, \\; u_k ∈ \\mathcal{U} \\; \\forall k.
+```
+
+### Fields
+
+- `M` -- mass matrix
+- `C` -- viscosity matrix
+- `K` -- stiffness matrix
+- `B` -- input matrix
+- `d` -- affine term
+- `X` -- state constraints
+- `U` -- input constraints
+"""
+SecondOrderConstrainedAffineControlDiscreteSystem
+
+for (Z, AZ) in ((:SecondOrderConstrainedAffineControlContinuousSystem, :AbstractContinuousSystem),
+                (:SecondOrderConstrainedAffineControlDiscreteSystem, :AbstractDiscreteSystem))
+    @eval begin
+        struct $(Z){T, MTM <: AbstractMatrix{T},
+                       MTC <: AbstractMatrix{T},
+                       MTK <: AbstractMatrix{T},
+                       MTB <: AbstractMatrix{T},
+                       VT <: AbstractVector{T},
+                       ST,
+                       UT} <: $(AZ)
+            M::MTM
+            C::MTC
+            K::MTK
+            B::MTB
+            d::VT
+            X::ST
+            U::UT
+            function $(Z)(M::MTM, C::MTC, K::MTK, B::MTB, d::VT, X::ST, U::UT) where {T, MTM <: AbstractMatrix{T},
+                                                                                  MTC <: AbstractMatrix{T},
+                                                                                  MTK <: AbstractMatrix{T},
+                                                                                  MTB <: AbstractMatrix{T},
+                                                                                  VT <: AbstractVector{T},
+                                                                                  ST,
+                                                                                  UT}
+                @assert checksquare(M) == checksquare(C) == checksquare(K) == size(B, 1) == length(d)
+                return new{T, MTM, MTC, MTK, MTB, VT, ST, UT}(M, C, K, B, d, X, U)
+            end
+        end
+        function $(Z)(M::Number, C::Number, K::Number, B::Number, d::Number, X, U)
+             return $(Z)(hcat(M), hcat(C), hcat(K), hcat(B), [d], X, U)
+        end
+
+        statedim(s::$Z) = size(s.C, 1)
+        inputdim(::$Z) = size(s.B, 2)
+        noisedim(::$Z) = 0
+        mass_matrix(s::$Z) = s.M
+        viscosity_matrix(s::$Z) = s.C
+        stiffness_matrix(s::$Z) = s.K
+        affine_term(s::$Z) = s.d
+        input_matrix(s::$Z) = s.B
+        stateset(s::$Z) = s.X
+        inputset(s::$Z) = s.U
+    end
+    for T in [Z, Type{<:eval(Z)}]
+        @eval begin
+            islinear(::$T) = true
+            isaffine(::$T) = true
+            ispolynomial(::$T) = false
+            isnoisy(::$T) = false
+            iscontrolled(::$T) = false
```

Comment:
```suggestion
            iscontrolled(::$T) = true
```
same for `SecondOrderConstrainedAffineControlContinuousSystem`(?)